gratipay/utils/__init__.py

```diff
@@ -133,45 +133,26 @@ def get_team(state):
     return team
 
 
-def b64encode_s(s):
-    prefix = b''
-    if not isinstance(s, bytes):
-        s = s.encode('utf8')
-    else:
-        # Check whether the string is binary or already utf8
-        try:
-            s.decode('utf8')
-        except UnicodeError:
-            prefix = b'.'
-    return prefix + b64encode(s, b'-_').replace(b'=', b'~')
+def encode_for_querystring(s):
+    """Given a unicode, return a unicode that's safe for transport across a querystring.
+    """
+    if not isinstance(s, unicode):
+        raise TypeError('unicode required')
+    return urlsafe_b64encode(s.encode('utf8')).replace(b'=', b'~').decode('ascii')
 
 
-def b64decode_s(s, **kw):
-    def error():
+def decode_from_querystring(s, **kw):
+    """Given a unicode computed by encode_for_querystring, return the inverse.
+    """
+    if not isinstance(s, unicode):
+        raise TypeError('unicode required')
+    try:
+        return urlsafe_b64decode(s.encode('ascii').replace(b'~', b'=')).decode('utf8')
+    except:
```

Comment:
This is a little confusing. Why does `decode_from_querystring` require `unicode` if it only ends up encoding it into `ascii`? Doesn't that mean, it can only process `ascii`?